.circleci/config.yml

```diff
@@ -8,6 +8,29 @@ references:
       - image: aetrnty/builder
         user: builder
     working_directory: ~/epoch
+    environment:
+      DOCKER_CLIENT_VERSION: "17.09.0-ce"
+      DOCKERHUB_REPO: aetrnty/epoch
+
+  setup_remote_docker: &setup_remote_docker
+    setup_remote_docker:
+      docker_layer_caching: true
+      version: "17.09.0-ce"
+
+  install_docker_client: &install_docker_client
+    run:
+      name: Install Docker client
+      command: |
+        curl -L -o /tmp/docker-${DOCKER_CLIENT_VERSION:?}.tgz \
+          https://download.docker.com/linux/static/stable/x86_64/docker-${DOCKER_CLIENT_VERSION:?}.tgz
+        tar -xz -C /tmp -f /tmp/docker-${DOCKER_CLIENT_VERSION:?}.tgz
+        mkdir ~/bin && mv /tmp/docker/docker ~/bin
+
+  docker_build: &docker_build
+    run:
+      name: Build Docker image
+      command: |
+        ~/bin/docker build -t $DOCKERHUB_REPO:$DOCKER_TAG .
```

Comment:
Why the need for tagging it at all? No uploading is done so it looks like no tagging needed.

Also, I wonder if any state is persisted between runs so there is clash in tag name? (that BTW is branch name)

Reply:
I'll tag and push in the following pull requests, anyway I removed the obsoleted tagging (in scope of this PR) in 0d1f9069 for the sake of atomicity

Reply:
thanks